tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -106,488 +95,446 @@ describe(__filename, () => {
+    const { params } = signInUserWithProps({ biography: '' });
 
-    const root = renderUserProfile({ params, store });
+    renderUserProfile({ params });
 
-    expect(root.find('.UserProfile-biography')).toHaveLength(0);
+    expect(screen.queryByText('Biography')).not.toBeInTheDocument();
   });
 
   it('does not render a report abuse button if user is the current logged-in user', () => {
-    const root = renderUserProfile();
+    signInUserWithProps({ userId: 100 });
+    renderUserProfile();
 
-    expect(root.find(ReportUserAbuse)).toHaveLength(0);
+    expect(
+      screen.queryByRole('button', { name: 'Report this user for abuse' }),
+    ).not.toBeInTheDocument();
   });
 
   it('renders a report abuse button if user is not logged-in', () => {
-    const { store } = dispatchClientMetadata();
-    const root = renderUserProfile({ store });
+    renderUserProfile();
 
-    expect(root.find(ReportUserAbuse)).toHaveLength(1);
+    expect(
+      screen.getByRole('button', { name: 'Report this user for abuse' }),
+    ).toBeInTheDocument();
   });
 
   it('renders a report abuse button if user is not the current logged-in user', () => {
     const userId = 1;
-    const { store } = dispatchSignInActions({ userId });
+    signInUserWithProps({ userId });
 
     // Create a user with another userId.
     const user = createUserAccountResponse({ id: 222 });
     store.dispatch(loadUserAccount({ user }));
 
     // See this other user profile page.
     const params = { userId: user.id };
-    const root = renderUserProfile({ params, store });
+    renderUserProfile({ params });
 
-    expect(root.find(ReportUserAbuse)).toHaveLength(1);
+    expect(
+      screen.getByRole('button', { name: 'Report this user for abuse' }),
+    ).toBeInTheDocument();
   });
 
   it('still renders a report abuse component if user is not loaded', () => {
     // The ReportUserAbuse handles an empty `user` object so we should
     // always pass the `user` prop to it.
-    const root = renderUserProfile({ params: { userId: 123 } });
+    renderUserProfile({ params: { userId: 123 } });
 
-    expect(root.find(ReportUserAbuse)).toHaveLength(1);
+    expect(
+      screen.getByRole('button', { name: 'Report this user for abuse' }),
+    ).toBeInTheDocument();
   });
 
   it('renders two AddonsByAuthorsCard', () => {
-    const root = renderUserProfile();
+    renderUserProfile();
 
-    expect(root.find(AddonsByAuthorsCard)).toHaveLength(2);
-  });
-
-  it('passes the errorHandler to the AddonsByAuthorsCard', () => {
-    const errorHandler = createStubErrorHandler();
-    const root = renderUserProfile({ errorHandler });
-
-    expect(root.find(AddonsByAuthorsCard).at(0)).toHaveProp(
-      'errorHandler',
-      errorHandler,
-    );
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'errorHandler',
-      errorHandler,
-    );
-  });
-
-  it('passes the errorHandler to Page', () => {
```

Comment:
Because `it('renders errors')` tests the error string occurs 3 times it does _kinda_ test that the errors are being passed down because that's where some of the other errors are being rendered, but not in a reliable way.

Reply:
https://github.com/mozilla/addons-frontend/issues/11363 but not sure what the fix would look like

Reply:
Ah, I missed the fact that it is asserting the error string shows up 3 times. As I said, I think I've looked at this code too much. The odd thing is that 3 times could be once for `UserProfile` and twice for the two `AddonsByAuthorsCard`s, but what about `Page`? I guess that can be investigated via the follow-up you filed.